### PR TITLE
Uplift third_party/tt-metal to c0eb6a589efbca3684382059bce66331b6402c27 2025-07-11

### PR DESCRIPTION
**Pull request by vmilosevic**
This PR uplifts the third_party/tt-metal to the c0eb6a589efbca3684382059bce66331b6402c27

- Update calls to ::tt::tt_metal::allocate_tensor_on_device after metal commit 4f000593b1
- Remove references to MultiDeviceHostStorage after removed in metal commit 5ba9bd798a
- Increase trace region size for trace test

### CODE REVIEW

**Review by nsmithtt — APPROVED** [review]

**Comment by brataTT** [comment]
<details><summary>Brings 115 metal commits</summary>

```j
> git log cf17101cb79329ee9369b2ab3075ef8a22c00008..c0eb6a589efbca3684382059bce66331b6402c27 --format="%cd %h by %an|||: %s" --date=short-local | column -t -s "|||"
2025-07-11 c0eb6a589e by David Ma                   : Remove watcher singleton, move the server under MetalContext (#24730)
2025-07-10 01c7bf9026 by Kannika Kabilar            : Comment out the failing yolov8x tests from the Single-card device perf pipeline (#24901)
2025-07-10 1b9f02d307 by Kannika Kabilar            : Comment out the failing tests from the Single-card device perf pipeline (#24861)
2025-07-10 5ba9bd798a by Oleg Milyutin              : #15840: Unify `HostStorage` and `MultiDeviceHostStorage` (#24656)
2025-07-10 746225adca by Umair Bilal Cheema         : tt-fabric 2D Mcast implementation (#24312)
2025-07-10 e6ddac588e by Sudhanshu Singhal          : creating tensor rather than torch.cat (#24737)
2025-07-10 5f766780e2 by John Bauman                : Fix TSAN error creating ethernet kernel (#24862)
2025-07-10 e8a26d353d by Radomir Djogo              : Rd/reduce w fp32 (#24665)
2025-07-10 a81dd9f372 by Rui Zhang                  : Add methods in HAL to get read/write alignment (#24808)
2025-07-10 27ca667d07 by Oleg Milyutin              : [skip ci] #22746: Add a high-level architecture document for multi-host mesh runtime (#24796)
2025-07-10 74cdf6c309 by Nigel Huang                : Enable FD on Fabric for N300 and T3K (#24341)
2025-07-10 1a67031f8a by Juan Camilo Vega           : Correcting CCL core placement for full TG compatibility (#24844)
2025-07-10 dae714ab0d by William Ly                 : Revert "#22199: Move ttnn gather from experimental" (#24875)
2025-07-10 904cd80729 by Bryan Wilder Field Lozano  : [skip ci] Restart SSH Daemon in container when restarted (#24841)
2025-07-10 8c34474584 by William Ly                 : [skip ci] Add P100, P150b to Metal L2 choices (#24871)
2025-07-10 6547c324b1 by Miłosz Gajewski            : #22199: Move ttnn gather from experimental (#24684)
2025-07-10 8ae120bbf3 by Andrew Fuller              : Bump ClangTidyLight's timeout (#24865)
2025-07-10 9d5993a9dc by Adrian Morrison            : MoE A2A Combine handling local reduction (#24797)
2025-07-10 9cccd0d5f8 by Gustavo Sarabando          : [produce data] [skip ci] Remove redundant api calls (#24801)
2025-07-10 d3b812a82e by Stanislav Minakov          : #24356: Update python definition of create_sharded_memory_config (#24819)
2025-07-10 4f4b550f42 by William Ly                 : Revert "Fix big buffers (4GB+) overflow in ttnn codebase and metal slow dispatch" (#24869)
2025-07-10 c97c4db512 by Saad Jameel                : Optimize all_to_all_dispatch by reducing the number of packets send and add Fabric 1D support (#24816)
2025-07-10 c1fc891587 by William Ly                 : Use CIv2 P150b runners (#24804)
2025-07-10 25f6b56c0d by Patrick Roberts            : #21967: Support 5D broadcast (#24321)
2025-07-10 c297669413 by Punith Sekar               : Update swin_v2 and yolov6l model name in fast_dispatch_full_regressions yaml file (#24753)
2025-07-10 64e15809d3 by Abhishek Agarwal           : [Fabric] Updated test infra [2/n] (#24559)
2025-07-10 8f1882e364 by Mouliraj Elamurugan        : #24787: Update silu op (#24820)
2025-07-10 adc6985434 by Ridvan Song                : Removing reliance on ethernet connection for location (#24305)
2025-07-10 35cd25343a by Miłosz Gajewski            : #22198: Uint32 support for ttnn.sort index tensor added (#23403)
2025-07-10 8257775ae4 by Nikola Cvetkovic           : #24175: Revert z and y dim to default in uninit (#24694)
2025-07-10 a93aab6fe1 by Grayson Rechsteiner        : Add Sharding Support To Minimal AG & RS (#24659)
2025-07-10 eed9b3c2ab by Radoica Draskic            : Remove unused rot matrix tensors from `tt_transformers` (#24835)
2025-07-10 f580efeee5 by Pratikkumar Prajapati      : MOE: Add support of prefill for the Expert layer (#24679)
2025-07-10 88c771ca15 by Evan Banerjee              : moving tt_metal tests from the metalium debian to the metalium-validation debian (#24789)
2025-07-10 b701d724a7 by Bartłomiej Cieślar         : Implement the DeepSeek Example MLP (#24837)
2025-07-10 f4dc344f3d by Pavlo Hilei                : Fix big buffers (4GB+) overflow in ttnn codebase and metal slow dispatch (#22731)
2025-07-10 8ad0783241 by Radoica Draskic            : Add `max_seq_len` as parameter to vllm interface (#24836)
2025-07-10 52a605b79f by Juan Camilo Vega           : Fixing synchronization in CCLs for BH compatibility (#24733)
2025-07-10 df93ece9d8 by Sankar Manoj               : #0: Skip Conv DRAM UTs on Blackhole (#24780)
2025-07-10 d91ff58982 by Mark O'Connor              : Update Programming_Mesh_of_Devices_with_TT-NN.md (#24833)
2025-07-10 d2a0e76220 by Djordje Ivanovic           : Update T3K Falcon7b perf targets (#24782)
2025-07-10 e476ee2370 by Iva Potkonjak              : SDXL: binary gen set for UNet (#24757)
2025-07-10 2cea0e15df by Mark O'Connor              : Adjust perf bounds for Qwen T3K CI tests (#24716)
2025-07-10 5761f50cc9 by Punith Sekar               : Yolov6l trace+2cq (#24629)
2025-07-09 11fcc418d1 by Jason Davies               : [skip ci] Fix typos in {INSTALLING,METALIUM_GUIDE}.md. (#24762)
2025-07-09 73f14ad4ad by Andrew Fuller              : Remove incomplete check (#24814)
2025-07-09 eded79103b by Samarth Agarwal            : Implement support for sharded buffers with page sizes > max prefetch command size (#24675)
2025-07-09 71c0f8c6c9 by Andrew Fuller              : Clang Tidy Light for PRs (#24788)
2025-07-09 b9a48d018a by Nigel Huang                : Partial Revert keep multi device fixtures open (#24795)
2025-07-09 e45ab7653d by Joseph Chu                 : Add simple `tt-run` MPI launcher for distributed applications (#24647)
2025-07-09 faff4e569d by Oleg Milyutin              : [skip ci] Add `unit_tests_ttnn_tensor` to t3k tests (#24779)
2025-07-09 dfa359a0ac by Sean Nijjar                : patch - recover pipeline health (#24792)
2025-07-09 18fb31d4c8 by John Bauman                : Use mask to avoid processing unused CBs in firmware (#22551)
2025-07-09 981d22f5c2 by Paytin Gardner             : Adding missing int to int typecasts (#24257)
2025-07-09 0160fdb7a5 by Adrian Morrison            : MoE all_to_all_combine follow-up (#24639)
2025-07-09 67ac0d55e6 by Adrian Morrison            : Slice (Split) program cache bug (#24667)
2025-07-09 8936a3e222 by Evan Banerjee              : removing unused variables in tt_metal (#24409)
2025-07-09 e4964be8b7 by Nigel Huang                : Compile command queue kernels in parallel (#24597)
2025-07-09 af9b52d452 by Allan Liu                  : Add generic support for ND routing tables (#24549)
2025-07-09 818d1253a2 by Pavle Petrovic             : Unit offset for RMSNorm (#24518)
2025-07-09 b24976f332 by Bryan Wilder Field Lozano  : Refine memcpy.hpp (#24691)
2025-07-09 6c2da71bd5 by Almeet Bhullar             : Update dispatch bw filter script (#24555)
2025-07-09 07f00e5d67 by Sohaib Nadeem              : Generate topology.json file to store routing information from control plane when NoC tracing is enabled (#24560)
2025-07-09 9085b0e5c9 by Keerthana Rajendran        : #13272: Vanilla unet: Trace2CQ (#24299)
2025-07-09 a40fbd0684 by Stuti Raizada              : Llama TG: async to_torch processing (#24707)
2025-07-09 8fe3da9fb4 by Yu Gao                     : #0: fixes for dram sharded matmul (#24714)
2025-07-09 b1535519a4 by Miłosz Gajewski            : #22934: data movements optimization for ttnn sort (#24572)
2025-07-09 6293238bf8 by Sofija Jovic               : Avoid pool l1 usage assert in no-dispatch mode (#24701)
2025-07-09 cb2b55a2dd by Djordje Ivanovic           : Revert "Fix import of minimal reduce scatter test" (#24748)
2025-07-09 16a2344e3f by Harry Andrews              : #24697: remove incorrect memory config (#24700)
2025-07-09 a85a35bd31 by Aswin Zayasankaran         : Add int32 support for neg (#24605)
2025-07-09 cc833cd487 by Sankar Manoj               : #20810: Add support for tiled inputs and outputs to slice_write (#23780)
2025-07-09 ef808bbf8a by Aswin Zayasankaran         : chore: update LLK submodule to 7d865f9 (#24720)
2025-07-09 78a7184929 by Djordje Ivanovic           : Revert "Minimal line reduce scatter" (#24746)
2025-07-09 173e2bb783 by Pavlo Hilei                : Fix accessor tests and benchmark (#24624)
2025-07-09 ccfb1c2793 by Raymond Kim                : #24545: Use tech debt sidedoor to add additional TTT dependencies into LLMBox image and enable its offline use (#24699)
2025-07-09 37078661b7 by Anasuya G Nair             : Migrate addalpha, subalpha to device (#24431)
2025-07-08 dbe2678c65 by Kannika Kabilar            : Disable yolov8x and squeezebert from (Single-card) Device perf pipeline by pytest skip (#24739)
2025-07-08 7508b4dc85 by Stanislav Minakov          : Add API to get consumed memory per bank from TensorSpec (#24676)
2025-07-08 54454f4b13 by Venkatesh Guduru           : #24252: move SentenceBERT post-process to ttnn (#24507)
2025-07-08 f940d47106 by Marvin Mok                 : Initial: Added test infra to support DRAM tests (#24496)
2025-07-08 9f7cfa2ddf by Stanislav Minakov          : Fix build with a different python version in a virtual environment (#24677)
2025-07-08 7fad3e4e63 by Oleg Milyutin              : #24631: Extend mesh coord infra to compute neighbors along any dims. (#24670)
2025-07-08 4f000593b1 by Oleg Milyutin              : #24447: `allocate_tensor_on_host`, `copy_device_to_host_tensor` APIs (#23865)
2025-07-08 1b2c66f074 by Radoica Draskic            : Change data-parallel batch size calculation in tt_transformers (#24683)
2025-07-08 b897448abd by Brian Beggs                : [skip ci] Create Installing_Troubleshooting_Galaxy_WH_6U.md, test/debug guide fo… (#22890)
2025-07-08 0dc508adc8 by Ammar Vora                 : Add support for larger head_dim in paged_update_cache (#24632)
2025-07-08 6c89d00d41 by Sabira Shaik               : Restructure trace pipeline for yolov8x: Updated folder Structure, filenames, and method names (#24565)
2025-07-08 727746ac2b by Venkatesh Guduru           : #13772: add yolov11 performant demo (#24203)
2025-07-08 4b504b24fc by Sudharsan Vijayaraghavan   : #19340: Swin_v2_s ttnn model bringup (#19901)
2025-07-08 744fd046d7 by Naif Tarafdar              : Remove print in slice (#24724)
2025-07-08 5d17d2d62b by Colman Glagovich           : Fix import of minimal reduce scatter test (#24734)
2025-07-08 8be4005e62 by Austin Ho                  : Uplift umd to get updated get_ethernet_connections_to_remote_devices functionality (#24594)
2025-07-08 f834267261 by Rui Zhang                  : Remove using std::string from tt-metalium/utils.hpp (#24416)
2025-07-08 6106484c0f by Raymond Kim                : #0: Do not check compile time anymore for e2e perf models because we couldn't give less of a shit (#24601)
2025-07-08 14c2293dcf by Austin Ho                  : #24668: Fix call to memcpy to not read extra bytes (#24674)
2025-07-08 04e4bc8d89 by Saad Jameel                : Add 6U tests, a new TG perf pipeline and perf tests for all_to_all_dispatch (#24641)
2025-07-08 ef4801c645 by Colman Glagovich           : Minimal line reduce scatter (#24188)
2025-07-08 0ffcd40af7 by Raymond Kim                : #0: [skip ci] Regress functional unet perf with trace from 2400 fps to 2300 fps (#24719)
2025-07-08 03ac7d08bf by Sean Nijjar                : Enable Erisc IRAM by default when persistent fabric is launched for Wormhole (#20523)
2025-07-08 2949982589 by Punith Sekar               : Fix yolov6l CI failure (#24638)
2025-07-08 7ce7a11dae by Evan Banerjee              : fixing unused variables in ttnn version 3 (#24646)
2025-07-08 a674d77889 by Juan Camilo Vega           : Removing int32 tests that cause an ND PCC issue due to an LLK bug (#24705)
2025-07-08 8a11efa33c by Aliaksei Sala              : Add layout argument for arange operation (#24491)
2025-07-08 59192e8bba by Bryan Wilder Field Lozano  : [skip ci] Update merge-gate.yaml to give extra second of tolerance (#24709)
2025-07-08 ed33141b81 by Aswin Zayasankaran         : Add int32 support for ge and le (#23991)
2025-07-08 c48046d6dd by Vishal Chaudhary           : #23591: Move gelu implementation to llk repo (#23996)
2025-07-08 22a3f7ec18 by Marina Vasiljevic          : Fix llama prefill test for 6U (#24685)
2025-07-08 239d1a7d9e by Mouliraj Elamurugan        : #23171: Migrate atanh as adevice op  (#23709)
2025-07-08 326eb066d3 by Nathan Maurice             : Make assert_with_ulp() robust to empty tensors, NaN and Inf values (#23533)
2025-07-08 78f7f3c495 by Johanna Rock               : Revert: Topk optimisation: indices as optional input tensor (#24434) (#24573)
2025-07-08 7e23da2714 by Nilaykumar Patel           : Enable folding with pad for conv2d folds. (#23525)
2025-07-08 60477de55d by Marko Staletovic           : Updated tests for the single core upsample operation (#24608)
2025-07-08 31d70fab6d by Djordje Ivanovic           : Llama70b - fix perf unit tests (#24649)
2025-07-08 db41476abc by Bartłomiej Cieślar         : Separate the _new_state into an externally called create_state (#24622)
```

</details> 

FE CI runs:
- [x] [tt-torch CI](https://github.com/tenstorrent/tt-torch/actions/runs/16218091291) 
  - `llama_3b` PCC loss to `0.96`. Test will be updated in tt-torch.
- [x] [tt-forge-fe CI](https://github.com/tenstorrent/tt-forge-fe/actions/runs/16218091858)  
  - repeat op tests strict XPASS reported as fail
  - `efficientnet` PCC loss to `0.989`. Test need to be updated.
- [x] [tt-xla CI](https://github.com/tenstorrent/tt-xla/actions/runs/16217444944)
